domain/infosync/info.go

```diff
@@ -364,7 +371,67 @@ func UpdatePlacementRules(ctx context.Context, rules []*placement.RuleOp) error
 	}
 
 	_, err = doRequest(ctx, addrs, path.Join(pdapi.Config, "rules/batch"), http.MethodPost, bytes.NewReader(b))
-	return err
+	if err != errPlacementRulesDisabled {
+		return err
+	}
+	return nil
+}
+
+// GetAllRuleBundles is used to get all rule bundles from PD. It is used to load full rules from PD while fullload infoschema.
+func GetAllRuleBundles(ctx context.Context) ([]*placement.Bundle, error) {
+	is, err := getGlobalInfoSyncer()
+	if err != nil {
+		return nil, err
+	}
+
+	bundles := []*placement.Bundle{}
+	if is.etcdCli == nil {
+		return bundles, nil
+	}
+
+	addrs := is.etcdCli.Endpoints()
+
+	if len(addrs) == 0 {
+		return nil, errors.Errorf("pd unavailable")
+	}
+
+	res, err := doRequest(ctx, addrs, path.Join(pdapi.Config, "placement-rule"), "GET", nil)
+	if err == nil {
+		err = json.Unmarshal(res, &bundles)
+	}
+	if err != errPlacementRulesDisabled {
+		return bundles, err
+	}
+	return bundles, nil
```

Comment:
It seems that `GetAllRuleBundles` and `GetRuleBundle` have duplicated code, only the `bundles` setting is different. Could we extract this code into a function?

Reply:
The latest commit is the best way I can think of.